src/main/java/com/zombiecastlerush/util/Prompter.java

```diff
@@ -18,12 +18,59 @@ static String getUserInput(String displayMessage , String... args) {
         return result;
     }
 
-    static void movePlayer(Player player) {
-        List<Room> availableRooms = player.getCurrentPosition().getConnectedRooms();
-        String moveString = Prompter.getUserInput("\nYou are in " + player.getCurrentPosition() + "\n\nTo go somewhere, please type go and one of the following available locations: " + availableRooms);
-        List<String> moveList = Parser.parse(moveString);
-        if (moveList != null) {
-            player.moveTo(moveList.get(1));
+    static void controller(Player player) {
+        Room currentRoom = player.getCurrentPosition();
+        List<Room> availableRooms = currentRoom.getConnectedRooms();
+        List<Item> currRoomInventory = currentRoom.inventory.getItems();
+
+        System.out.println("You are in " + currentRoom + ". "+ currentRoom.getDescription());
+        if(currentRoom.getChallenge()!=null && !currentRoom.getChallenge().isCleared()){
+            String currRoomChallenge = currentRoom.getChallenge().getDescription();
+            System.out.println("The room has " + currRoomChallenge + " and " + currRoomInventory);
+            System.out.println("After the "+ currRoomChallenge +" is solved, you can go to one of the following available locations: " + availableRooms);
+        }
+        else {
+            System.out.println("The room has " + currRoomInventory+"\nYou can go to one of the following locations " + availableRooms);
+        }
+
+        String userInput = Prompter.getUserInput("\nEnter \"help\" if you need help with the commands");
+        List<String> userInputList = Parser.parse(userInput);
+        if (userInputList!=null && userInputList.size()==2) {
+            String action =userInputList.get(0);
+            switch (action){
+                case "go":
```

Comment:
Since we're reusing "go" and "attempt" from ALLOWED_ACTIONS, perhaps we should make this an Enum.

Reply:
change it to enum later PR